lib/smart_proxy_container_gateway/container_gateway_api.rb

```diff
@@ -19,6 +19,28 @@ class Api < ::Sinatra::Base
       inject_attr :database_impl, :database
       inject_attr :container_gateway_main_impl, :container_gateway_main
 
+      get '/index/static/?' do
+        # TODO: filter out repositories that are not tied to the (optional) authenticated host
+        # host = <lookup host>
+        # catalog = host_catalog(host)
+
+        # pulp_response = container_gateway_main.flatpak_static_index(translated_headers_for_proxy, params)
+        # if pulp_response.code.to_i >= 400
+        #   status pulp_response.code.to_i
+        #   body pulp_response.body
+        # end
+
+        # pulp_index = JSON.parse(pulp_response.body)
+        # pulp_index["Results"] = pulp_index["Results"].select { |result| catalog.include?(result["Name"]) }
```

Comment:
I wonder if it's cheaper to inline the change
```suggestion
        # pulp_index["Results"].select! { |result| catalog.include?(result["Name"]) }
```

Reply:
Yeah, that should be a bit better.